SonarQube.TeamBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -124,40 +126,44 @@ private bool DoExecute(ProcessedArgs args, ILogger logger)
 
         #region Private methods
 
-        private void InstallLoaderTargets(ProcessedArgs args, ILogger logger)
+        private void InstallLoaderTargets(ProcessedArgs args)
         {
             if (args.InstallLoaderTargets)
             {
-                this.targetInstaller.InstallLoaderTargets(logger);
+                this.targetInstaller.InstallLoaderTargets(this.logger);
             }
             else
             {
-                logger.LogDebug(Resources.MSG_NotCopyingTargets);
+                this.logger.LogDebug(Resources.MSG_NotCopyingTargets);
             }
         }
 
-        private bool FetchArgumentsAndRulesets(ProcessedArgs args, TeamBuildSettings settings, ILogger logger, out IDictionary<string, string> serverSettings)
+        private bool FetchArgumentsAndRulesets(ProcessedArgs args, TeamBuildSettings settings, out IDictionary<string, string> serverSettings)
         {
             string hostUrl = args.GetSetting(SonarProperties.HostUrl);
             serverSettings = null;
 
-            ISonarQubeServer server = this.serverFactory.Create(args, logger);
+            ISonarQubeServer server = this.serverFactory.Create(args, this.logger);
             try
             {
                 // Fetch the SonarQube project properties
-                logger.LogInfo(Resources.MSG_FetchingAnalysisConfiguration);
+                this.logger.LogInfo(Resources.MSG_FetchingAnalysisConfiguration);
                 serverSettings = server.GetProperties(args.ProjectKey);
 
                 // Generate the FxCop rulesets
-                logger.LogInfo(Resources.MSG_GeneratingRulesets);
-                GenerateFxCopRuleset(server, args.ProjectKey, "csharp", "cs", "fxcop", Path.Combine(settings.SonarConfigDirectory, FxCopCSharpRuleset), logger);
-                GenerateFxCopRuleset(server, args.ProjectKey, "vbnet", "vbnet", "fxcop-vbnet", Path.Combine(settings.SonarConfigDirectory, FxCopVBNetRuleset), logger);
+                this.logger.LogInfo(Resources.MSG_GeneratingRulesets);
+                GenerateFxCopRuleset(server, args.ProjectKey, "csharp", "cs", "fxcop", Path.Combine(settings.SonarConfigDirectory, FxCopCSharpRuleset));
+                GenerateFxCopRuleset(server, args.ProjectKey, "vbnet", "vbnet", "fxcop-vbnet", Path.Combine(settings.SonarConfigDirectory, FxCopVBNetRuleset));
+
+                SonarLintAnalyzerProvider.SetupAnalyzers(server, settings, args.ProjectKey, this.logger);
+
+                CompilerAnalyzerConfig analyzerConfig = this.analyzerProvider.SetupAnalyzers(server, settings, args.ProjectKey);
```

Comment:
This is the only other change in the file: a call to the new analyser provider. Still more to do: need to remove the call to the old SonarLintAnalyzerProvider once the new provider is finished.